contracts/token/erc721/abstract/ImmutableERC721Base.sol

```diff
@@ -216,6 +222,26 @@ abstract contract ImmutableERC721Base is
         _totalSupply--;
     }
 
+    /// @dev Burn a token, checking the owner of the token against the parameter first.
+    function safeBurn(address owner, uint256 tokenId) public virtual {
+        address currentOwner = ownerOf(tokenId);
+        if (currentOwner != owner) {
+            revert IImmutableERC721MismatchedTokenOwner(tokenId, currentOwner);
+        }
+
+        burn(tokenId);
+    }
+
+    /// @dev Burn a batch of tokens, checking the owner of each token first.
+    function safeBurnBatch(IDBurn[] memory burns) public virtual {
```

Comment:
can we expose this on the preset instead?

Reply:
Done.